sdk/eventhub/event-hubs/src/managementClient.ts

```diff
@@ -304,33 +308,78 @@ export class ManagementClient extends LinkEntity {
-        maxRetries: options.retryOptions && options.retryOptions.maxRetries,
-        abortSignal: options.abortSignal,
-        requestName: options.requestName,
-        timeoutInSeconds: getRetryAttemptTimeoutInMs(options.retryOptions) / 1000,
-        delayInSeconds:
-          options.retryOptions &&
-          options.retryOptions.retryInterval &&
-          options.retryOptions.retryInterval >= 0
-            ? options.retryOptions.retryInterval / 1000
-            : undefined
+      const sendOperationPromise = () =>
+        new Promise<Message>(async (resolve, reject) => {
+
+          let waitTimer: any;
+          const actionAfterTimeout = () => {
+            const address = this._mgmtReqResLink!.receiver.address || "address";
+            const desc: string =
+              `The request with message_id "${request.message_id}" to "${address}" ` +
+              `endpoint timed out. Please try again later.`;
+            const e: Error = {
+              name: "OperationTimeoutError",
+              message: desc
+            };
+            return reject(translate(e));
+          };
+
+          waitTimer = setTimeout(
+            actionAfterTimeout,
+            getRetryAttemptTimeoutInMs(options!.retryOptions)
+          );
+
+          const initOperationStartTime = Date.now();
+          log.mgmt(
+            "[%s] Acquiring lock to get the management req res link.",
+            this._context.connectionId
+          );
+          await defaultLock.acquire(this.managementLock, () => {
+            return this._init().catch((err: Error) => {
+              clearTimeout(waitTimer);
+              reject(translate(err));
+            });
+          });
+          const initOperationEndTime = Date.now();
+
+          if (!options) {
+            options = {};
+          }
+
+          const operationTimeoutInMs = initOperationEndTime - initOperationStartTime;
+          const sendRequestOptions: SendRequestOptions = {
+            abortSignal: options.abortSignal,
+            requestName: options.requestName,
+            timeoutInSeconds: operationTimeoutInMs / 1000
+          };
+          resolve((await this._mgmtReqResLink!.sendRequest(request, sendRequestOptions)));
+        });
+
+      const jitterInSeconds = randomNumberFromInterval(1, 4);
+      const maxRetries = options.retryOptions && options.retryOptions.maxRetries;
+      const delayInSeconds =
+        options.retryOptions &&
+        options.retryOptions.retryInterval &&
+        options.retryOptions.retryInterval >= 0
+          ? options.retryOptions.retryInterval / 1000
+          : Constants.defaultDelayBetweenOperationRetriesInSeconds;
+      const config: RetryConfig<void> = {
+        operation: sendOperationPromise,
+        connectionId: this._context.connectionId,
+        operationType:
+          request.to && request.to === Constants.cbsEndpoint
+            ? RetryOperationType.cbsAuth
+            : RetryOperationType.management,
```

Comment:
Since the code here is being used by the managementClient only, the `operationType` can only be `RetryOperationType.management`

Reply:
Oh, do have issues for tracking other changes in SDK that rely on `core-amqp`'s sendRequest()?

Reply:
The `RequestResponseLink` class is used in the below places
- cbs link in core-amqp which is used for negotiating claims when a new AMQP sender/receiver/management link needs to be created
     - Since #2835 ensures that the the act of link creation itself is now being retried, we don't have to retry the act of negotiating claims. 
     - The only thing we need to ensure is that if claim negotiation fails, then that error bubbles all the way up to the place where we create the link using init(). It would be good to have a test for this if possible.
- management link in Event Hubs and Service Bus
     - Event Hubs is being taken care of this PR. I'll ensure there is an issue for Service Bus, when I create the plan for Track 2